Correct ghost zones to Patch 1.6  sub-areas, remove invalid graveyards

FlagFlayer opened the pull request:
## 🍰 Pullrequest
Currently, when you die outside of a starting area, your ghost can still be sent to the graveyard in the starting area should it be the closest graveyard to the player character on death. This pull request corrects the ghost zones of graveyards in starting areas to use the ghost zones of the starting areas, instead of using the ghost zone for the entire zone.

### Proof
"Starting area graveyards are still reserved exclusively for characters of similar faction that die **within their confines**. For example, the spirit of a Horde character dying right outside of [Deathknell](https://wowpedia.fandom.com/wiki/Deathknell) will not use Deathknell graveyard, but will instead go to the next closest Tirisfal Glades graveyard." (Patch 1.6 change notes: https://web.archive.org/web/20060104092012/http://www.worldofwarcraft.com/patchnotes/patch-05-08-02.html)

TBC Classic behaviour recorded by @heyitsbench : https://www.youtube.com/watch?v=wyaBOODuwqs&t=53s

### Issues
- Fixes https://github.com/vmangos/core/issues/1417

### How2Test
- Make a horde/alliance character
- Die in a location such that the closest graveyard is the graveyard in the starting area, but the character is not in the starting area zone.

### Todo / Checklist
<!-- In case some parts are still missing, important notes, breaking changes and other notable items, list them here. -->
- [X] None

FlagFlayer committed:
Correct graveyard zones for starting area graveyards

FlagFlayer committed:
Update 20220922195507_world.sql

FlagFlayer committed:
Update 20220922195507_world.sql

FlagFlayer commented:
For Gates of Ironforge Graveyard: "An Alliance-only graveyard has been added just outside of Ironforge, around the side of the mountain. It is only for use by Alliance characters that die inside Ironforge and **in the Gates of Ironforge sub-area of [Dun Morogh](https://vanilla-wow-archive.fandom.com/wiki/Dun_Morogh)**. Horde characters dying therein will continue to only use the Kharanos graveyard."

For Caer Darrow: A neutral graveyard has been added on the island of Caer Darrow in Western Plaguelands. It will serve characters who die on **Caer Darrow** and in Scholomance.

FlagFlayer commented:
Missing implementation for:

Blackwood Lake: "A neutral graveyard has been added in the Eastern Plaguelands near the northwestern shore of Blackwood Lake. **Characters that die in The Fungal Vale will continue to use the Darrowshire graveyard due to the mountains between it and the new graveyard.** Additionally, characters dying in Stratholme will now use the new Blackwood Lake graveyard."

Crown Guard Tower (patch 1.12.1): "Players will now port to the Crown Guard Tower graveyard if they die in Fungal Vale **while their faction controls the tower.**"

If anyone knows how to implement these, it can be done in a seperate pull request. For now these fixes can be merged.

Daribon commented:
I just tested your SQL and it breaks all starting zones graveyards. If I die in Coldridge Valley with your SQL applied, I will get sent to Kharanos. Similar issue with Northshire and the other starting zone graveyards.

--------------

About the The Fungal Vale problem:
If you set  The Fungal Vale 2258 to use graveyard 634 Darrowshire you will be sent to Blackwood.
Because 909 Eastern Plaguelands, Blackwood Lake using area entry for the entire zone and 909 being actually closer by air.

SQL to test:
```UPDATE `game_graveyard_zone` SET `ghost_zone`='2258' WHERE  `id`=634 AND `ghost_zone`=139;```

Edit:
Might be some bug with having gys use specific area ids that are part of a zone. It only seems possible to use area ids that are for entire zones.

Daribon committed:
Fix remaining GY issues

GY 911 is Raven Hill, should have been 869 for Caer Darrow.
469 is not a neutral Darnassus GY. It is Darkshore, Twilight Vale GY.
Added missing GY for Coldridge Pass.
Added Darrowshire GY to The Fungal Vale.

Daribon commented:
Issue I described above has been fixed since this commit https://github.com/vmangos/core/commit/d65953dfcf18063b6f6f06f953a652a26296fe85

I've fixed some stuff on the PR:
> GY 911 is Raven Hill, should have been 869 for Caer Darrow.
469 is not a neutral Darnassus GY. It is Darkshore, Twilight Vale GY.
Deleted Darkshore, Twilight Vale GY for Stonetalon Mountains.
Added missing Dun Morogh, Anvilmar GY for Coldridge Pass.
Added Darrowshire GY to The Fungal Vale.

Only one query was working and being correct when I started testing this PR.
It is now all fixed. Ready to be merged.

----

Added Crown Guard Tower graveyard to The Fungal Vale area in this PR: https://github.com/vmangos/core/pull/1636

ratkosrb committed:
Remove quotes from integers.
